api/src/main/java/org/pmiops/workbench/ras/RasOidcClientConfig.java

```diff
@@ -0,0 +1,34 @@
+package org.pmiops.workbench.ras;
+
+import static org.pmiops.workbench.ras.RasLinkConstants.AUTHORIZE_URL_SUFFIX;
+import static org.pmiops.workbench.ras.RasLinkConstants.RAS_SECRET_BUCKET_NAME;
+import static org.pmiops.workbench.ras.RasLinkConstants.TOKEN_URL_SUFFIX;
+import static org.pmiops.workbench.ras.RasLinkConstants.USER_INFO_URL_SUFFIX;
+
+import java.io.IOException;
+import javax.inject.Provider;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.google.CloudStorageClient;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Lazy;
+
+@Configuration
+public class RasOidcClientConfig {
+  static final String RAS_OIDC_CLIENT = "RAS_OIDC_CLIENT";
+
+  @Bean(RAS_OIDC_CLIENT)
+  @Lazy
+  public OpenIdConnectClient rasOidcClient(
+      CloudStorageClient cloudStorageClient, Provider<WorkbenchConfig> workbenchConfigProvider)
+      throws IOException {
+    String rasClientSecret = cloudStorageClient.getCredentialsBucketString(RAS_SECRET_BUCKET_NAME);
+    String rasClientId = workbenchConfigProvider.get().ras.clientId;
+    String rasTokenUrl = workbenchConfigProvider.get().ras.host + TOKEN_URL_SUFFIX;
+    String rasAuthorizeUrl = workbenchConfigProvider.get().ras.host + AUTHORIZE_URL_SUFFIX;
+    String rasUserInfoUrl = workbenchConfigProvider.get().ras.host + USER_INFO_URL_SUFFIX;
+
+    return new OpenIdConnectClient(
```

Comment:
Is this client safe for reuse, or concurrent use? If not, this should be `@RequestScope` (in general, this may be safer)

Reply:
Note: the default behavior for spring beans is **singleton**.

Reply:
I think this should be singleton, and better to be singleton. The main thing is `HttpTransport`, and I think that is a heavy-weight one. Their javadoc recommandation is: 
"For maximum efficiency, applications should use a single globally-shared instance of the HTTP transport.

Reply:
btw, do you know why my api-local-test  are failing for some related `OpenIdConnectClient` scope error?

Reply:
Nice, it is directly related to this discussion. The WorkbenchConfig value is request scoped (some discussion of that here: https://pmi-engteam.slack.com/archives/GP4CG4TFF/p1575930937326700?thread_ts=1575925918.323100&cid=GP4CG4TFF). The CloudStorageService depends on the config, therefore this injection will not work since your bean is singleton. Changing this to request scoped would be the easiest fix.

RE: HttpTransport, is there an alternate constructor that accepts that parameter? We already have a bean for that which you inject, and is already a singleton: https://github.com/all-of-us/workbench/blob/f56ff491ee1f303d434a89290e86dc944e6db0a4/api/src/main/java/org/pmiops/workbench/config/AppEngineConfig.java#L12

Reply:
Thanks ,changed to request scope.  HttpTransport, did a code search, seems most examples use that as a static final variable.

Reply:
Injection seems preferred since we already even have a bean to do this. If we wanted to write a unit test for OpenIdConnectClient we would immediately need to deal with this static dependency